File: convert_to_traditional.py
import os
from opencc import OpenCC

# Initialize OpenCC for Simplified to Traditional conversion
converter = OpenCC('s2t')

def convert_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as file:
        content = file.read()
    
    # Convert content
    converted_content = converter.convert(content)

    # Write back to the file
    with open(file_path, 'w', encoding='utf-8') as file:
        file.write(converted_content)
    print(f"Converted: {file_path}")

def traverse_and_convert(directory):
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith('.md'):
                file_path = os.path.join(root, file)
                convert_file(file_path)

# Start conversion from the current directory
if __name__ == "__main__":
    traverse_and_convert(os.getcwd())

File: convertToTraditional.js
const fs = require('fs');
const path = require('path');
const OpenCC = require('opencc');

const converter = new OpenCC('s2t.json');

function convertDirectory(directory) {
    fs.readdir(directory, (err, files) => {
        if (err) {
            console.error(`無法讀取目錄 ${directory}:`, err);
            return;
        }

        files.forEach(file => {
            const filePath = path.join(directory, file);
            fs.stat(filePath, (err, stats) => {
                if (err) {
                    console.error(`無法讀取檔案 ${filePath}:`, err);
                    return;
                }

                if (stats.isDirectory()) {
                    convertDirectory(filePath);
                } else if (stats.isFile() && path.extname(file) === '.md') {
                    fs.readFile(filePath, 'utf8', (err, data) => {
                        if (err) {
                            console.error(`無法讀取檔案 ${filePath}:`, err);
                            return;
                        }

                        converter.convertPromise(data)
                            .then(traditionalText => {
                                fs.writeFile(filePath, traditionalText, 'utf8', err => {
                                    if (err) {
                                        console.error(`無法寫入檔案 ${filePath}:`, err);
                                    } else {
                                        console.log(`已轉換: ${filePath}`);
                                    }
                                });
                            })
                            .catch(err => {
                                console.error(`轉換失敗 ${filePath}:`, err);
                            });
                    });
                }
            });
        });
    });
}

convertDirectory(__dirname);